src/test/libcephfs/test.cc

```diff
@@ -984,12 +984,23 @@ TEST(LibCephFS, PreadvPwritev) {
   struct iovec iovin[2];
   struct iovec iovout[2];
   ssize_t nwritten, nread;
+  char *out0, *out1, *in0, *in1; 
 
+  out0 = new char[str0.size()];
+  out1 = new char[str1.size()];
+  in0 = new char[str0.size()];
+  in1 = new char[str1.size()];
+
+  iovout[0].iov_base = out0;
```

Comment:
btw, i think it's not necessary to `strcpy` the input data, instead you can reuse the buffer in `str0` and `str1`: simply pointing the `iovin[0].iov_base` to `str0.data()` would do it.